docs/client.rst

```diff
@@ -747,6 +747,36 @@ If no redirects occurred or ``allow_redirects`` is set to ``False``,
 history will be an empty sequence.
 
 
+Client signals
+---------------
+
+The execution flow of a specific request can be followed attaching listeners functions
+to the signals provided by the :class:`ClientSession` instance, for example the following
+snippet shows how the start and the end of the request flow can be followed::
+
+    async def on_request_start(trace_context, method, host, port, headers):
+        print("Starting request")
+
+    async def on_request_end(trace_context, resp):
+        print("Ending request")
```

Comment:
Wording bit: is `on_request_end` happens just before request is ended or when it's actually ended. `-ing` suffix confusing here.